### DISCUSSION TITLE
How to explain the information from a visualized uplift tree ?

### DISCUSSION THREAD

**Discussion by 00helloworld** [uber/causalml#528] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/42757757/177490148-398e2014-c738-4db9-9a1b-f57511eb103f.png)
Hi, I'm confused about the `uplift score` showed in the visualized tree.
In `uplift_tree_visualization.ipynb` it said that _uplift score: the treatment effect between treatment and control (when there are multiple treatment groups, this is the maximum of the treatment effects)_. So I think the uplift score of the leaf node in this picture should be `0.63-0.44=0.19`. But the `uplift score` in the picture is `0.2458`.
How to explain that and is my calculation right or not?
Thanks

**Comment by jeongyoonlee** [comment, marked as answer]
cc: @zhenyuz0500, @t-tte, @paullo0106

**Reply by 00helloworld** [reply]
Thanks a lot! Waiting for some explanation.

**Reply by paullo0106** [reply]
we might need to take a closer look into [UpliftTree](https://github.com/uber/causalml/blob/master/causalml/inference/tree/uplift.pyx#L891), the uplift score in the tree plot is what we have in `nodeSummary`  but `results` saves something different and it's used in the last row in the plot for the mean of each class.  

some other notes
- This issue doesn't seem to happen to the root node. 
- This gap seems to be there for a while, as shown in some of the early uplift tree plots

<img width="1199" alt="Screen Shot 2022-08-29 at 4 39 47 PM" src="https://user-images.githubusercontent.com/4966393/187317586-83e24f53-8841-4291-83c7-ed61939e4230.png">


cc: @jeongyoonlee

**Reply by paullo0106** [reply]
the gap comes from [n_reg](https://github.com/uber/causalml/blob/master/causalml/inference/tree/uplift.pyx#L857), the regularization parameter where we can also make it consistent to avoid confusion in the uplift tree visualization

for example, the numbers would look aligned if we pass in `n_reg = 0`  (default is 100)
![Screen Shot 2022-08-29 at 6 06 32 PM](https://user-images.githubusercontent.com/4966393/187325868-16835d5f-fed6-45bf-a67e-523ee07083b4.png)

**Reply by 00helloworld** [reply]
Thanks a lot!!!! 
That's so helpful ！
